### ISSUE TITLE
Bug: path-keys-no-trailing-slash is triggered on path key of /

### ISSUE THREAD

**Issue by MikeRalphson** [stoplightio/spectral#79]
### **I'm submitting a...**
  - [x] bug report

### What is the current behavior?

The linter warning `path-keys-no-trailing-slash` is triggered on a `path` key of `/`which is incorrect as all `path` keys MUST start with a `/` character.

```
src/cli/bin/run lint .minimal.yaml
```

where `minimal.yaml` is:

```yaml
openapi: 3.0.0
info:
  title: ''
  version: ''
paths:
  /: {}
```

Results:

```
 ›   Warning: { name: 'path-keys-no-trailing-slash',
 ›      summary: 'given keys should not end with a slash.',
 ›      message: 'must not match the pattern \'/$\'',
 ›      path: [ 'paths', '/' ],
 ›      severity: 40,
 ›      severityLabel: 'warn' }
```

### What is the expected behavior?

This warning is **not** triggered.

### What is the motivation / use case for changing the behavior?

Eliminating spurious warnings.

### Please tell us about your environment:

  - Version: 0.0.0
  - Language: [TypeScript 3.2.2]

### Other information

`oas-kit` (the source of the default rules in `Speccy`, and hence the inspiration for a number of rules in `spectral`) uses an `omit` parameter to the `pattern` rule to chomp off the leading `/` character before comparison. Doubtless this could be done within the regex by someone with the appropriate skills.

**Comment by tbarn** [comment]
Yup, that's a bug! I'll see if I can break out my rusty regex skills and fix it. If not, I'll pass it to someone else on the team for help. 

If anyone else wants to take it, let me know!
